src/corehost/cli/deps_resolver.cpp

```diff
@@ -164,6 +164,18 @@ void deps_resolver_t::setup_shared_store_probes(
     }
 }
 
+pal::string_t deps_resolver_t::get_probe_directories()
+{
+    pal::string_t directories;
+    for (const auto& pc : m_probes)
```

Comment:
This change does not look correct. The goal of this change is to list the additional probing paths. m_probes contains all sort of probing locations (store, FX - which is flat, servicing location, etc). Additional probing path is one of the various kind of probe locations - see https://github.com/dotnet/core-setup/blob/master/src/corehost/cli/deps_resolver.cpp#L202

I think you should be filtering out additional probing paths and only adding them here. Also, the property APP have is that they have a nuget layout, which is not true for all other probing locations.